src/phpDocumentor/ApiReference/Factory.php

```diff
@@ -47,25 +62,15 @@ public function create(DocumentGroupDefinitionInterface $definition)
             throw new InvalidArgumentException('Definition must be an instance of ' . DocumentGroupDefinition::class);
         }
 
-        // TODO: move this to a custom factory so that we can inject the factory in this class and then have the
-        // middleware as dependency of that factory.
+        $strategies = $this->stategies;
+        $strategies[] = new File(
```

Comment:
Using such a factory will also remove the need for injecting the middleware into this class; and that will improve the SRP part of this class